conformance/tests/backendtlspolicy.go

```diff
@@ -152,5 +157,112 @@ var BackendTLSPolicy = suite.ConformanceTest{
+			ctx := t.Context()
+			routeNN := types.NamespacedName{Name: "backendtlspolicy", Namespace: ns}
+			gwNN := types.NamespacedName{Name: "same-namespace", Namespace: ns}
+			validconfigmap := types.NamespacedName{Name: "tls-checks-ca-certificate", Namespace: ns}
+
+			kubernetes.NamespacesMustBeReady(t, suite.Client, suite.TimeoutConfig, []string{ns})
+			gwAddr := kubernetes.GatewayAndRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), &gatewayv1.HTTPRoute{}, false, routeNN)
+			kubernetes.HTTPRouteMustHaveResolvedRefsConditionsTrue(t, suite.Client, suite.TimeoutConfig, routeNN, gwNN)
+
+			validPolicyNN := types.NamespacedName{Name: "normative-test", Namespace: ns}
+			kubernetes.BackendTLSPolicyMustHaveCondition(t, suite.Client, suite.TimeoutConfig, validPolicyNN, gwNN, acceptedCond)
+			kubernetes.BackendTLSPolicyMustHaveCondition(t, suite.Client, suite.TimeoutConfig, validPolicyNN, gwNN, resolvedRefsCond)
+
+			validcm := &corev1.ConfigMap{}
+			err := suite.Client.Get(ctx, validconfigmap, validcm)
+			require.NoErrorf(t, err, "failed to get valid configmap")
+
+			originalCAData := validcm.Data["ca.crt"]
+
+			t.Cleanup(func() {
+				cleanupCtx, cleanupCancel := context.WithTimeout(context.Background(),
+					suite.TimeoutConfig.DefaultTestTimeout)
+				defer cleanupCancel()
+
+				currentCM := &corev1.ConfigMap{}
+				err = suite.Client.Get(cleanupCtx, validconfigmap, currentCM)
+				if err != nil {
+					t.Logf("cleanup: error getting ConfigMap: %v", err)
+					return
+				}
+
+				if currentCM.Data["ca.crt"] != originalCAData {
+					restored := currentCM.DeepCopy()
+					restored.Data["ca.crt"] = originalCAData
+					err = suite.Client.Patch(cleanupCtx, restored, client.MergeFrom(currentCM))
+					if err != nil {
+						t.Logf("cleanup: error restoring ConfigMap: %v", err)
+					}
+				}
+			})
+			h.MakeRequestAndExpectEventuallyConsistentResponse(t, suite.RoundTripper, suite.TimeoutConfig, gwAddr,
+				h.ExpectedResponse{
+					Namespace: ns,
+					Request: h.Request{
+						Host: "abc.example.com",
+						Path: "/backendtlspolicy",
+					},
+					Response: h.Response{StatusCodes: []int{200}},
+				})
+
+			mutatedCM := validcm.DeepCopy()
+			mutatedCM.Data["ca.crt"] = ""
+			err = suite.Client.Patch(ctx, mutatedCM, client.MergeFrom(validcm))
+			require.NoErrorf(t, err, "failed to mutate ConfigMap")
+
+			invalidAcceptedCond := metav1.Condition{
+				Type:   string(gatewayv1.PolicyConditionAccepted),
+				Status: metav1.ConditionFalse,
+				Reason: string(gatewayv1.BackendTLSPolicyReasonNoValidCACertificate),
+			}
+			invalidResolvedRefsCond := metav1.Condition{
+				Type:   string(gatewayv1.BackendTLSPolicyConditionResolvedRefs),
+				Status: metav1.ConditionFalse,
+				Reason: string(gatewayv1.BackendTLSPolicyReasonInvalidCACertificateRef),
```

Comment:
I have tested this on Envoy Gateway, the test passes (apparently eg watches and reconciles correctly!). The only issue on envoy gateway is that this reason here is wrong there (it is getting us InvalidKind when per the GEP it should be `InvalidCACertificateRef`.)

@arkodg fyi